cpp/CMakeLists.txt

```diff
@@ -314,6 +333,7 @@ endif(BUILD_CUML_CPP_LIBRARY)
 # - include paths ---------------------------------------------------------------------------------
 
 include_directories(
```

Comment:
I think CMake is pushing the `target_*` interface, which specifies various parameters for each target, and distinguish `INTERFACE`, `BUILD` and `BUILD_INTERFACE` used for imported target.  It's less convenient but creates much better interface for someone else who want to use `find_package(cuml)` in their project.

Reply:
Not sure I got your point here. Can you elaborate @trivialfis ?

Reply:
@teju85 See https://cliutils.gitlab.io/modern-cmake/ and https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1 for some details.  I don't agree with all of them, but things like distinguishing BUILD and INTERFACE make sense to me.

Reply:
This is cool! There are some really good guidelines given in that Effective CMake talk. I have filed issue #1134 for a more organized cleanup and refactor of cmake. Thanks again @trivialfis